python-avd/pyavd/_eos_designs/structured_config/network_services/metadata.py

```diff
@@ -19,133 +20,91 @@ class MetadataMixin(Protocol):
-                    }
-                    for category in categories
-                    for application in get(category, "applications", default=[])
-                    if application["name"] in user_defined_app_names
-                ],
-            },
-        }
-
-        return strip_null_from_data(cv_pathfinder_metadata_applications, (None, [], {}))
+        for profile in get(self.application_traffic_recognition, "application_profiles", default=[]):
+            application_profile = EosCliConfigGen.Metadata.CvPathfinder.Applications.ProfilesItem(name=profile["name"])
+            protocols = get(profile, "application_transports")
+            if protocols:
+                application_profile.transport_protocols.extend(protocols)
+            for application in get(profile, "applications", default=[]):
+                if application["name"] not in user_defined_app_names:
+                    # TODO: Need to fix the support of services
+                    application_profile.builtin_applications.append(
+                        EosCliConfigGen.Metadata.CvPathfinder.Applications.ProfilesItem.BuiltinApplicationsItem(name=application["name"])
+                    )
+                if application["name"] in user_defined_app_names:
+                    application_profile.user_defined_applications.append(
+                        EosCliConfigGen.Metadata.CvPathfinder.Applications.ProfilesItem.UserDefinedApplicationsItem(name=application["name"])
+                    )
+            for category in get(profile, "categories", default=[]):
+                # TODO: Need to fix the support of services
+                application_profile.categories.append(EosCliConfigGen.Metadata.CvPathfinder.Applications.ProfilesItem.CategoriesItem(category=category["name"]))
+            self.structured_config.metadata.cv_pathfinder.applications.profiles.append(application_profile)
+        for category in categories:
+            for application in get(category, "applications", default=[]):
+                if application["name"] not in user_defined_app_names:
+                    self.structured_config.metadata.cv_pathfinder.applications.categories.builtin_applications.append(
+                        EosCliConfigGen.Metadata.CvPathfinder.Applications.Categories.BuiltinApplicationsItem(
+                            name=application["name"], category=category["name"]
+                        )
+                    )
+                    # TODO: Need to fix the support of services
+                if application["name"] in user_defined_app_names:
+                    self.structured_config.metadata.cv_pathfinder.applications.categories.user_defined_applications.append(
+                        EosCliConfigGen.Metadata.CvPathfinder.Applications.Categories.UserDefinedApplicationsItem(
+                            name=application["name"], category=category["name"]
+                        )
+                    )
```

Comment:
We can use `append_new` at lot of places here to make the code more concise

```suggestion
                    self.structured_config.metadata.cv_pathfinder.applications.categories.user_defined_applications.append_new(name=application["name"], category=category["name"]
                    )
```